pyomo/contrib/doe/fim_doe.py

```diff
@@ -0,0 +1,2774 @@
+        analysis_square.solve_time = time_solve1
+
+        if self.optimize:
+
+            m = self.__add_objective(m, deactive_obj=True)
+
+            # solve problem with DOF then
+            if self.verbose:
+                print('First solve with given objective:')
+            result_doe1 = self.__solve_doe(m, fix=True)
+
+            if self.verbose:
+                print('Second solve with given objective:')
+            time0_solve2 = time.time()
+            result_doe = self.__solve_doe(m, fix=False)
+            time1_solve2 = time.time()
+            time_solve2 = time1_solve2 - time0_solve2
+
+            # extract Jac
+            jac_optimize = self.__extract_jac(m)
+
+            # create result object
+            analysis_optimize = FIM_result(self.param_name, self.measure, jacobian_info=None, all_jacobian_info=jac_optimize,
+                                           prior_FIM=self.prior_FIM)
+            # for simultaneous mode, FIM and Jacobian are extracted with extract_FIM()
+            analysis_optimize.calculate_FIM(self.design_timeset, result=result_doe)
+            analysis_optimize.model = m
+
+            time1 = time.time()
+            # record optimization time
+            analysis_optimize.solve_time = time_solve2
+            analysis_optimize.total_time = time1-time0
+            if self.verbose:
+                print('Total solve time with simultaneous_finite mode (Wall clock) [s]:', time_solve1 + time_solve2)
+                print('Total wall clock time [s]:', time1-time0)
+
+            return analysis_square, analysis_optimize
+
+        else:
+
+            time1 = time.time()
+            # record square problem time
+            analysis_square.total_time = time1-time0
+            if self.verbose:
+                print('Total solve time with simultaneous_finite mode (Wall clock) [s]:', time_solve1)
+                print('Total wall clock time [s]:', time1 - time0)
+
+            return analysis_square
+
+    def compute_FIM(self, design_values, mode='sequential_finite', FIM_store_name=None, specified_prior=None,
+                    tee_opt=True, scale_nominal_param_value=False, scale_constant_value=1,
+                    store_output = None, read_output=None, extract_single_model=None,
+                    formula='central', step=0.001,
+                    objective_option='det'):
+        """This function solves a square Pyomo model with fixed design variables to compute the FIM.
+        It calculates FIM with sensitivity information from four modes:
+            1.  sequential_finite: Calculates a one scenario model multiple times for multiple scenarios. Sensitivity info estimated by finite difference
+            2.  sequential_sipopt: calculate sensitivity by sIPOPT [Experimental]
+            3.  sequential_kaug: calculate sensitivity by k_aug [Experimental]
+            4.  direct_kaug: calculate sensitivity by k_aug with direct sensitivity
```

Comment:
You're missing `simultaneous_finite` from this list of modes which is referenced above in the `__check_inputs` method and again near the bottom of this method.